pkgs/tools/misc/fastfetch/default.nix

```diff
@@ -0,0 +1,81 @@
+{ chafa
+, cmake
+, dbus
+, dconf
+, fetchFromGitHub
+, glib
+, imagemagick_light
+, lib
+, libglvnd
+, libxcb
+, makeWrapper
+, ocl-icd
+, opencl-headers
+, pciutils
+, pkg-config
+, stdenv
+, vulkan-loader
+, wayland
+, xfce
+, xorg
+, zlib
+, enableChafa ? false
+, enableImageMagick ? false
+, enableOpenCLModule ? true
+, enableOpenGLModule ? true
+, enableVulkanModule ? true
+, enableWayland ? true
+, enableX11 ? true
+, enableXFCE ? false
+}:
+
+stdenv.mkDerivation rec {
+  pname = "fastfetch";
+  version = "1.5.6";
+
+  src = fetchFromGitHub {
+    owner = "LinusDierheimer";
+    repo = pname;
+    rev = version;
+    hash = "sha256-OsK78irfO9r6pSEmlN6mFoKhQZgBqIYO8syj5AHH4IE=";
+  };
+
+  nativeBuildInputs = [ cmake makeWrapper pkg-config ];
+
+  runtimeDependencies =
+    [
+      dbus
+      dconf
+      glib
+      pciutils
+      zlib
+    ]
+    ++ lib.optionals (enableChafa) [ chafa ]
+    ++ lib.optionals (enableImageMagick) [ imagemagick_light ]
+    ++ lib.optionals (enableOpenCLModule) [ ocl-icd ]
+    ++ lib.optionals (enableOpenGLModule) [ libglvnd ]
+    ++ lib.optionals (enableVulkanModule) [ vulkan-loader ]
+    ++ lib.optionals (enableWayland) [ wayland ]
+    ++ lib.optionals (enableX11) [ libxcb ]
+    ++ lib.optionals (enableXFCE) [ xfce.xfconf ];
+
+  buildInputs =
+    runtimeDependencies
+    ++ lib.optionals (enableOpenCLModule) [ opencl-headers ]
+    ++ lib.optionals (enableX11) [ xorg.libX11 ];
+
+  LD_LIBRARY_PATH = lib.makeLibraryPath runtimeDependencies;
```

Comment:
I'd rename this to something like `libraryPath`, the `makeWrapper` below ends up being a bit weird to read now i think

Reply:
> I'd rename this to something like `libraryPath`, the `makeWrapper` below ends up being a bit weird to read now i think

Why? If `LD_LIBRARY_PATH` is the name of an environment variable commonly used for build tools, why change it? I think that the UPPERCASE makes variables like this more clear; it tells me that this is not an option. Just curious.

Reply:
Is there a reason not to:
```nix
  postInstall = let
    ldLibraryPath = lib.makeLibraryPath runtimeDependencies;
  in ''
    wrapProgram $out/bin/fastfetch --prefix LD_LIBRARY_PATH : "${ldLibraryPath}"
    wrapProgram $out/bin/flashfetch --prefix LD_LIBRARY_PATH : "${ldLibraryPath}"
  '';
```

Am I wrong to assume it is best practice to keep things as local as possible?

Reply:
Only reason not to would be overrides. If its in the `mkDerivation` scope you could do this to add something to the generated wrapper:

```nix
fastpatch.overrideAttrs (prev: {
  libraryPath = prev.libraryPath or [] ++ [ some_lib ];
})
```
While if its local to the phase you'd have to rewrite the whole thing because you cannot append to the list.

Because of that I'd say it's best to keep stuff like dependencies in the `mkDerivation` scope, and use the `let ... in` for values that have little reason to be overwritten. Stuff like string shorthands you otherwise need to repeat, quick functions you'll use once, etc.